cpp/models/lct_secir_2_diseases/model.h

```diff
@@ -0,0 +1,732 @@
+        // Must be calculated separately in order not to overwrite the already calculated values ​​for Recovered.
+        // for 2a
+        flow = (FP)LctStateGroup::template get_num_subcompartments<InfectionState::InfectedCritical_2a>() *
+               (1 / params.template get<TimeInfectedCritical_a<FP>>()[Group]) *
+               y[ICri_2a_first_index +
+                 LctStateGroup::template get_num_subcompartments<InfectionState::InfectedCritical_2a>() - 1];
+        dydt[ICri_2a_first_index +
+             LctStateGroup::template get_num_subcompartments<InfectionState::InfectedCritical_2a>() - 1] -= flow;
+        dydt[Ri_ab] = dydt[Ri_ab] + (1 - params.template get<DeathsPerCritical_a<FP>>()[Group]) * flow;
+        dydt[Di_a]  = dydt[Di_a] + params.template get<DeathsPerCritical_a<FP>>()[Group] * flow;
+        // for 2b
+        flow = (FP)LctStateGroup::template get_num_subcompartments<InfectionState::InfectedCritical_2b>() *
+               (1 / params.template get<TimeInfectedCritical_b<FP>>()[Group]) *
+               y[ICri_2b_first_index +
+                 LctStateGroup::template get_num_subcompartments<InfectionState::InfectedCritical_2b>() - 1];
+        dydt[ICri_2b_first_index +
+             LctStateGroup::template get_num_subcompartments<InfectionState::InfectedCritical_2b>() - 1] -= flow;
+        dydt[Ri_ab] = dydt[Ri_ab] + (1 - params.template get<DeathsPerCritical_b<FP>>()[Group]) * flow;
+        dydt[Di_b]  = dydt[Di_b] + params.template get<DeathsPerCritical_b<FP>>()[Group] * flow;
+
+        // Function call for next group if applicable.
+        if constexpr (Group + 1 < num_groups) {
+            get_derivatives_impl<Group + 1>(pop, y, t, dydt);
+        }
+    }
+
+    /**
+     * @brief Calculates flows that are caused by people becoming infected (outflow from compartment S, Ra or Rb) for Group1.
+     *
+     * This is done recursively by calculating the interaction terms with each group.
+     * @tparam Group1 The group for which the derivative of the compartment should be calculated. 
+     * @tparam Group2 The group that Group1 interacts with. 
+     * @param[in] pop The current state of the population in the geographic unit we are considering.
+     * @param[in] y The current state of the model (or a subpopulation) as a flat array.
+     * @param[in] t The current time.
+     * @param[out] dydt A reference to the calculated output.
+     * The flow from Susceptible needs to be split into 2 parts for Exposed_1a and Exposed_1b:
+     * @param[out] part_a Reference to amount of flow caused by people infected with disease a.
+     * @param[out] part_b Reference to amount of flow caused by people infected with disease b.
+     * @param[in] compartment_index Index of the compartment for that the outflow will be calculated.
+     * @param[in] which_disease Index for which infected people cause the outflow (0 = a, 1 = b, 2 = a and b).
```

Comment:
```suggestion
     * @param[in] relevant_disease Index for which infected people cause the outflow (0 = a, 1 = b, 2 = a and b).
```
I think that makes the meaning a bit clearer.